TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row test failure

**Describe the problem**

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/3361105?expandedTest=build%3A%28id%3A3361097%29%2Cid%3A192853

```
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row
--- progress: testdata/logic_test/regional_by_row: 44 statements/queries
E210826 10:14:01.305698 708061 jobs/adopt.go:410  [n1] 41  job 687808316928688129: adoption completed with error could not create unique constraint "uniq_idx"
--- progress: testdata/logic_test/regional_by_row: 50 statements/queries
E210826 10:14:03.320813 715057 jobs/adopt.go:410  [n1] 42  job 687808324227268609: adoption completed with error could not create unique constraint "uniq_idx"
--- progress: testdata/logic_test/regional_by_row: 82 statements/queries
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +testdata/logic_test/regional_by_row:914: SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 + WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 + OR message LIKE 'Scan%'
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 + ORDER BY ordinality ASC
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +expected:
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    Scan /Table/72/1/"@"/10/0
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    Scan /Table/72/1/"\xc0"/10/0
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    Scan /Table/72/1/"\x80"/10/0
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    output row: [10 10 11 12 NULL]
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +but found (query options: "rowsort" -> ignore the following ordering of rows) :
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    Scan /Table/72/1/"@"/10/0
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    Scan /Table/72/1/"\x80"/10/0
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
E210826 10:14:04.686596 689205 sql/logictest/logic.go:3560  [-] 43 +    output row: [10 10 11 12 NULL]
logic.go:2278:
testdata/logic_test/regional_by_row:914: SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
OR message LIKE 'Scan%'
ORDER BY ordinality ASC
expected:
Scan /Table/72/1/"@"/10/0
Scan /Table/72/1/"\xc0"/10/0
Scan /Table/72/1/"\x80"/10/0
fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
output row: [10 10 11 12 NULL]
but found (query options: "rowsort" -> ignore the following ordering of rows) :
Scan /Table/72/1/"@"/10/0
Scan /Table/72/1/"\x80"/10/0
fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
output row: [10 10 11 12 NULL]
--- done: testdata/logic_test/regional_by_row with config multiregion-9node-3region-3azs: 83 tests, 1 failures
E210826 10:14:04.687027 689205 sql/logictest/logic.go:3585  [-] 44  testdata/logic_test/regional_by_row:926: too many errors encountered, skipping the rest of the input
logic.go:3120:
testdata/logic_test/regional_by_row:926: error while processing
logic.go:3120: testdata/logic_test/regional_by_row:926: too many errors encountered, skipping the rest of the input
--- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row (9.77s)
```

**To Reproduce**

Seen in CI but so far I have not been able to reproduce this locally.

--- a/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row
+++ b/pkg/ccl/logictestccl/testdata/logic_test/regional_by_row
@@ -916,15 +916,15 @@ statement ok
 SET tracing = on,kv,results; SELECT * FROM regional_by_row_table WHERE pk = 10; SET tracing = off
 
 # If the row is not found in the local region, the other regions are searched
-# in parallel.
-query T rowsort
+# in parallel. Note that if the row is quickly found in one remote region, the
+# other remote region might not be read from at all.
+query T
 SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
  WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
- OR message LIKE 'Scan%'
+ OR message LIKE 'Scan%' AND message NOT LIKE 'Scan%"\\xc0"%'
  ORDER BY ordinality ASC
 ----
 Scan /Table/73/1/"@"/10/0
-Scan /Table/73/1/"\xc0"/10/0
 Scan /Table/73/1/"\x80"/10/0
 fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
 output row: [10 10 11 12 NULL]

